sdk/appconfiguration/Azure.Data.AppConfiguration/src/SettingSelector.cs

```diff
@@ -17,58 +19,223 @@ namespace Azure.Data.AppConfiguration
     /// </summary>
     public class SettingSelector
     {
+        private StringBuilder _keys;
+        private StringBuilder _labels;
+        private SettingFields _fields = SettingFields.All;
+
+        private StringBuilder Keys => _keys ??= new StringBuilder();
+        private StringBuilder Labels => _labels ??= new StringBuilder();
+
+        private const char AnyChar = '*';
+        private const char EscapeChar = '\\';
+        private const char SeparatorChar = ',';
+        private const string KeyQueryFilter = "key";
+        private const string LabelQueryFilter = "label";
+        private const string FieldsQueryFilter = "$select";
+
         /// <summary>
         /// A wildcard that matches any key or any label when passed as a filter
         /// to Keys or Labels.
         /// </summary>
         public static readonly string Any = "*";
 
         /// <summary>
-        /// Keys or key filters that will be used to select a set of <see cref="ConfigurationSetting"/> entities.
+        /// Indicates the point in time in the revision history of the selected <see cref="ConfigurationSetting"/> entities to retrieve.
+        /// If set, all properties of the <see cref="ConfigurationSetting"/> entities in the returned group will be exactly what they
+        /// were at this time.
         /// </summary>
-        /// <remarks>See the documentation for this client library for details on the format of filter expressions.</remarks>
-        public IList<string> Keys { get; }
+        public DateTimeOffset? AcceptDateTime { get; set; }
 
         /// <summary>
-        /// Labels or label filters that will be used to select a set of <see cref="ConfigurationSetting"/> entities.
+        /// Creates a <see cref="SettingSelector"/> that will retrieve all <see cref="ConfigurationSetting"/> entities in the configuration store.
         /// </summary>
-        /// <remarks>See the documentation for this client library for details on the format of filter expressions.</remarks>
-        public IList<string> Labels { get; }
+        public SettingSelector() { }
 
         /// <summary>
-        /// The fields of the <see cref="ConfigurationSetting"/> to retrieve for each setting in the retrieved group.
+        /// Creates a <see cref="SettingSelector"/> that will retrieve <see cref="ConfigurationSetting"/> entities that match the passed-in key filter.
         /// </summary>
-        public SettingFields Fields { get; set; } = SettingFields.All;
+        /// <param name="rawKeyFilter">A key filter indicating which <see cref="ConfigurationSetting"/> entities to select.</param>
+        public SettingSelector(string rawKeyFilter)
```

Comment:
Why do the constructors take filters rather than literal key/label name?

Reply:
I'm currently thinking that it would be better to replace those with static factory method, something like `SettingSelector.FromFilterString`.